localstack/services/s3/s3_listener.py

```diff
@@ -666,6 +666,25 @@ def return_response(self, method, path, data, headers, response):
             response.status_code = 204
             return response
 
+        # emulate ErrorDocument functionality if a website is configured
+
+        if method == 'GET' and response.status_code == 404 and parsed.query != 'website':
+            s3_client = aws_stack.connect_to_service('s3')
+
+            try:
+                # Verify the bucket exists in the first place--if not, we want normal processing of the 404
+                s3_client.head_bucket(Bucket=bucket_name)
```

Comment:
Is the additionaly `head_bucket` request needed? Perhaps it could be removed. We would then simply call `get_bucket_website`, proceed if the call fails in `except`, and also proceed if no error document is configured.

Reply:
I did this `head_bucket` call to work around an issue I couldn't figure out. What I saw was if I tried to `get_bucket_website` is that 404s where the bucket had no website configured would take much longer to resolve to a 404. From what I could tell, something was trying to retry the connection repeatedly before throwing an exception, but trying to configure the s3_client to have max_attempts of 1 didn't help. Any thoughts on what would have worked better?

Reply:
I see, that makes sense. `boto3` has some automated retries built in, which are hard-coded depending on the response code. In that case your solution makes sense - thanks for clarifying.